### PR TITLE
Support for formatting Slovenian

### PR DESCRIPTION
**Pull request by filips123**
This adds support for formatting Slovenian numbers, dates, times and durations. It seems to work on all examples and tests I tested.

But there is one small problem that `nice_duration` only supports basic plural form while Slovenian uses [many plural forms](https://unicode-org.github.io/cldr-staging/charts/37/supplemental/language_plural_rules.html#sl). I was able to fix this in other functions, but `nice_duration` relies on plural files  (`day.word` and `days.word`, etc.), so there will be some wrong plural forms in some cases. However, it should still be understandable for people. I might check how I could fix this in future separate PR (it might require rewriting `nice_duration` plural support to something that works with more languages).

I will probably also create separate parsing PR in the future.

I have already signed CLA but it is may not be approved yet.

### CODE REVIEW

**Comment by devs-mycroft** [comment]
Hello, @filips123, thank you for helping with the Mycroft project! We welcome everyone
into the community and greatly appreciate your help as we work to build an AI
for Everyone.

To protect yourself, the project, and users of Mycroft technologies we require
a Contributor Licensing Agreement (CLA) before accepting any code
contribution. This agreement makes it crystal clear that along with your
code you are offering a license to use it within the confines of this project.
You retain ownership of the code, this is just a license.

Please visit https://mycroft.ai/cla to initiate this one-time signing. Thank
you!

**Comment by ChanceNCounter** [comment]
`nice_duration` is one of the trickier functions. For a first pass, I absolutely wouldn't worry about complex cases; the LF team isn't likely to call a language "fully supported" for *quite* a while after initial work has been done.

After all, the upcoming release is v0.2.3, and a pending refactor will make it 0.3.0 👅 so "slow and steady" is still the name of the game in *most* languages.

At any rate, LF's a small team, so code review might take a little while. But I can see that you're localizing the right things, in what looks like the right way, and you've had success *using* the functions, so I'm gonna seize the initiative and give you your Hacktoberfest credit. I imagine @krisgesling will be along to speak to the CLA, and anything else that might need saying from Mycroft-the-company's perspective (I'm not an employee, just a collaborator on this package.)

**Comment by krisgesling** [comment]
CLA received, and you've been added to our [list of excellent humans](https://github.com/mycroftAI/contributors), thanks!

**Comment by ChanceNCounter** [comment]
Alright. I've [rebased this](https://github.com/filips123/lingua-franca/compare/format-slovenian...ChanceNCounter:format-slovenian) for compatibility with LF 0.3+

@filips123 if you wouldn't mind checking out that branch (ChanceNCounter/format-slovenian) and ensuring that everything works, you can --force push it on top of this PR branch (I didn't wanna force-push someone else's repo)

It might need some touching up to conform with @JarbasAl's unified formatting and locations, I haven't looked yet. I've just done the basic rebase and integration.

**Comment by filips123** [comment]
Thanks! I tested it and everything seems to work correctly. I also have an idea how to improve plural support in `nice_duration` and I will probably create separate PR for this next month (because it will also be useful for other languages).

**Comment by ChanceNCounter** [comment]
@krisgesling @JarbasAl Now having worked on this PR a little, I'd appreciate another look before it gets merged. Also, it might not be conformant with Jarbas' new layout. The localizer's finding everything, though, so either Jarbas is magic, everything's in the right place, or both.

File: lingua_franca/res/text/sl-si/normalize.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{
+  "lowercase": false,
+  "numbers_to_digits": true,
+  "expand_contractions": true,
```
Comment (JarbasAl) [inline lingua_franca/res/text/sl-si/normalize.json:4]: this is True but there is nothing in the contractions dict bellow?

File: test/test_format_sl.py [hunk 2]
```diff
@@ -0,0 +1,567 @@
...
+                dt = datetime.datetime(i, 1, 31, 13, 2, 3)
+                self.assertTrue(len(nice_year(dt, lang=lang)) > 0)
+                # Looking through the date sequence can be helpful
+
+#                print(nice_year(dt, lang=lang))
+
+    def test_nice_duration(self):
+        self.assertEqual(nice_duration(1), "ena sekunda")
+
+        # This should be "sekunde" but this is unsupported
```
Comment (JarbasAl) [inline test/test_format_sl.py:529]: can you start this comment with TODO , that allows searching, and most IDEs support this, that way future devs know this needs work

File: test/test_format_sl.py [hunk 3]
```diff
@@ -0,0 +1,567 @@
...
+                         "osemnajstmilijonti")
+        self.assertEqual(pronounce_number(18e12, ordinals=True,
+                                          short_scale=False),
+                         "osemnajstbilijonti")
+        self.assertEqual(pronounce_number(18e12, ordinals=True),
+                         "osemnajsttrilijonti")
+        self.assertEqual(pronounce_number(18e18, ordinals=True,
+                                          short_scale=False), "osemnajsttrilijonti")
+
+# def nice_time(dt, lang="en-us", speech=True, use_24hour=False,
```
Comment (JarbasAl) [inline test/test_format_sl.py:290]: stray comment?

File: lingua_franca/lang/common_data_sl.py [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from collections import OrderedDict
+
+
+_ARTICLES_SL = {}
```
Comment (JarbasAl) [inline lingua_franca/lang/common_data_sl.py:19]: if articles exist they should be here, even if not used? any reason they are not replaced in normalize? eg, in portuguese and spanish articles are important and can not be removed in normalize, but they should still be present here

**Review by JarbasAl — CHANGES_REQUESTED** [review]
some nitpicks, but overall looks good

lingua_franca/lang/parse_sl.py is an empty file, can you at least add a comment, ```TODO implement parsing function```

**Comment by ChanceNCounter** [comment]
It can probably be addressed in another PR, but, at the moment, `parse.py` has to exist.

If it doesn't, this happens:

```
>>> from lingua_franca import parse
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/Chance/lingua-franca/lingua_franca/parse.py", line 33, in <module>
    populate_localized_function_dict("parse", langs=get_active_langs())
  File "/Users/Chance/lingua-franca/lingua_franca/internal.py", line 643, in populate_localized_function_dict
    mod = import_module(".lang." + lf_module + "_" + primary_lang_code,
  File "/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ModuleNotFoundError: No module named 'lingua_franca.lang.parse_sl'
```

whereas, with the empty file,

```
>>> lingua_franca.set_default_lang("sl")
>>> from lingua_franca import parse
>>> parse.extract_datetime("one")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/Chance/lingua-franca/lingua_franca/internal.py", line 592, in call_localized_function
    return _call_localized_function(func, *args, **kwargs)
  File "/Users/Chance/lingua-franca/lingua_franca/internal.py", line 550, in _call_localized_function
    raise loc_signature
lingua_franca.internal.FunctionNotLocalizedError: This function has not been implemented in the specified language.

**Comment by JarbasAl** [comment]
@ChanceNCounter i still think it's good for the file to exist, but with a TODO comment so devs can grep for TODO

File: lingua_franca/lang/common_data_sl.py (hunk 4 above)
Comment (filips123) [reply lingua_franca/lang/common_data_sl.py:19]: Slovenian has no articles. Most of the information available from articles in other languages (e.g. declensions, gender, definiteness, indefiniteness) can be determined from the word itself or accompanying adjectives, so I think nothing should be listed here.

**Review by JarbasAl — APPROVED** [review]

**Comment by ChanceNCounter** [comment]
Squashing manually to preserve authorship, then merging

**Pull request merged**